lib/cli/src/Cardano/CLI.hs

```diff
@@ -612,65 +638,71 @@ newCliKeyScheme = \case
     ledgerKeyFromSeed = Icarus.getKey
         . flip Icarus.generateKeyFromHardwareLedger pass
 
-    toPub (AXPrv xprv) = return . AXPub . CC.toXPub $ xprv
-    toPub (AXPub _) = Left "Input is already a public key."
-
-    derive
-        :: CC.DerivationScheme
-        -> XPrvOrXPub
-        -> DerivationIndex
-        -> Either String XPrvOrXPub
-    derive scheme1Or2 (AXPrv k) i =
-        return . AXPrv
-            $ CC.deriveXPrv
-                scheme1Or2
-                pass
-                k
-                (unDerivationIndex i)
-    derive scheme1Or2 (AXPub k) i =
-        maybe (Left err) (return . AXPub)
-            $ CC.deriveXPub
-                scheme1Or2
-                k
-                (unDerivationIndex i)
-      where
-        err = mconcat
-            [ T.unpack (toText i)
-            , " is a hardened index. Public key derivation is only possible for"
-            , " soft indices. \nIf the index is correct, please use the"
-            , " corresponding private key as input."
-            ]
+    -- We don't use passwords to encrypt the keys here.
+    pass = mempty
 
-    insp (AXPrv key) = do
-        let bytes = CC.unXPrv key
-        let (xprv, rest) = BS.splitAt 64 bytes
-        let (_pub, cc) = BS.splitAt 32 rest
-        let encodeToHex = T.pack . B8.unpack . hex
-        return $ mconcat
-            [ "extended private key: "
-            , encodeToHex xprv
-            , "\n"
-            , "chain code: "
-            , encodeToHex cc
-            ]
-    insp (AXPub key) = do
-        let bytes = CC.unXPub key
-        let (xpub, cc) = BS.splitAt 32 bytes
-        let encodeToHex = T.pack . B8.unpack . hex
-        return $ mconcat
+
+toPublic :: XPrvOrXPub -> Either String XPrvOrXPub
+toPublic (AXPrv xprv) = return . AXPub . CC.toXPub $ xprv
+toPublic (AXPub _) = Left "Input is already a public key."
+
+deriveChildKey
+    :: XPrvOrXPub
+    -> DerivationIndex
+    -> Either String XPrvOrXPub
+deriveChildKey (AXPrv k) i =
+    return . AXPrv
+        $ CC.deriveXPrv
+            CC.DerivationScheme2
+            (mempty :: Passphrase "Encryption")
```

Comment:
Hmmm. More like `mempty :: ScrubbedBytes` or `mempty :: ByteString` but, why not. Since `Passphrase` has a `ByteArrayAccess`, it probably doesn't matter.